[elastic] Added documentation missing description of required user roles permissions (#12498)
DataDog/integrations-core: elastic/README.md

      - To use the Agent's Elasticsearch integration for the AWS Elasticsearch services, set the `url` parameter to point to your AWS Elasticsearch stats URL.
      - All requests to the Amazon ES configuration API must be signed. See the [Making and signing OpenSearch Service requests][6] for details.
      - The `aws` auth type relies on [boto3][7] to automatically gather AWS credentials from `.aws/credentials`. Use `auth_type: basic` in the `conf.yaml` and define the credentials with `username: <USERNAME>` and `password: <PASSWORD>`.

>       - You have to create a user and a role (if you don't already have them) in Elasticsearch with the proper permissions to monitor. This can be done via the REST API offered by Elasticsearch or for example via Kibana UI. In the created role we must indicate the following properties:

Reviewer: ```suggestion
      - You must create a user and a role (if you don't already have them) in Elasticsearch with the proper permissions to monitor. This can be done through the REST API offered by Elasticsearch, or through the Kibana UI. Include the following properties in the created role:
```

        ```json
        name = "datadog"
        indices {
          names = [".monitoring-*", "metricbeat-*"]
          privileges = ["read", "read_cross_cluster", "monitor"]
        }
        cluster = ["monitor"]
        ```

>         And in the user indicate his roles:

Reviewer: ```suggestion
        Add the role to the user:
```

        ```json
        roles = [<created role>, "monitoring_user"]
        ```

>         For more information, see [create or update roles](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-role.html) and [create or update users](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-user.html).

Reviewer: ```suggestion
        For more information, see [create or update roles][29] and [create or update users][30].
```

Author: Done
